add statement of need + citations; closes #244 #243 (#266)
bcgov/bcdata: inst/JOSS/paper.Rmd

# Statement of Need

> `bcdata` is an R package that connects publicly available metadata and data sets in the British Columbia (B.C.) Data Catalogue to the diverse array of mapping, modeling and data processing capabilities of the R ecosystem. `bcdata` enables the efficient retrieval of British Columbia's geospatial data, and supports repeatable and reproducible analysis of hundreds of open-licensed British Columbia public sector data sets. By enabling programmatic access to the B.C. Data Catalogue using tidy data principles (@tidydata) and familiar R `tidyverse` syntax (@tidyverse), `bcdata` helps both novice and experienced R users find and use British Columbia government public and open data holdings.

Reviewer: ```suggestion
`bcdata` is an R package that connects publicly available metadata and data sets in the British Columbia (B.C.) Data Catalogue to the diverse array of mapping, modeling and data processing capabilities of the R ecosystem. `bcdata` enables the efficient retrieval of British Columbia's geospatial data, and supports repeatable and reproducible analysis of hundreds of open-licensed British Columbia public sector data sets. By enabling programmatic access to the B.C. Data Catalogue using familiar R `dplyr` syntax (@tidyverse), `bcdata` helps both novice and experienced R users find and use British Columbia government public and open data holdings.
```

# Introduction

> The British Columbia government hosts over 2000 tabular and geospatial data sets in the B.C. Data Catalogue (@bcdc).  Most provincial geospatial data is available through the B.C. Data Catalogue under an open licence, via a [Web Feature Service](https://en.wikipedia.org/wiki/Web_Feature_Service). A Web Feature Service is a powerful and flexible service for distributing geographic features over the web, supporting both geospatial and non-spatial querying.  The `bcdata` package for the R programming language (@RCore) wraps two distinct but complimentary web APIs - one for the B.C. Data Catalogue and one for the Web Feature Service.  This allows R users to search, download and import metadata and data from the B.C. Data Catalogue, as well as efficiently query and directly read geospatial data from the Web Feature Service into their R session. The `bcdata` package implements a novel application of `dbplyr` (@dbplyr) using a Web Feature Service backend---rather than a database backend---where a locally constructed query is processed by a remote server. This allows for fast and efficient geospatial data retrieval using familiar `dplyr` syntax. Through this functionality the `bcdata` package connects British Columbia government public data holdings in the B.C. Data Catalogue with the vast capabilities of R.

Reviewer: ```suggestion
The British Columbia government hosts over 2000 tabular and geospatial data sets in the B.C. Data Catalogue (@bcdc).  Most provincial geospatial data is available through the B.C. Data Catalogue under an open licence, via a [Web Feature Service](https://en.wikipedia.org/wiki/Web_Feature_Service). A Web Feature Service is a powerful and flexible service for distributing geographic features over the web, supporting both geospatial and non-spatial querying.  The `bcdata` package for the R programming language (@RCore) wraps two distinct but complimentary web APIs - one for the B.C. Data Catalogue and one for the Web Feature Service.  This allows R users to search, download and import metadata and data from the B.C. Data Catalogue, as well as efficiently query and directly read geospatial data from the Web Feature Service into their R session. The `bcdata` package implements a novel application of `dbplyr` (@dbplyr) using a Web Feature Service backend---rather than a database backend---where a locally constructed query is processed by a remote server. This allows for fast and efficient geospatial data retrieval while using `dplyr` syntax. Through this functionality the `bcdata` package connects British Columbia government public data holdings in the B.C. Data Catalogue with the vast capabilities of R.
```

  geom_sf(aes(fill = RANK_DESC))
```

> This demonstrates the efficiency of the filter-first then download approach: the size of the object downloaded by using `bcdc_query_geodata()` with `filter()` is `r paste0(format((1 - as.numeric(object.size(marmots2)) / as.numeric(object.size(all_sp_eco))) * 100, digits = 1), "%")` smaller than downloading the entire data set using `bcdcd_get_data()` and filtering locally.

Reviewer: ```suggestion
This demonstrates the efficiency of the filter-first then download approach: the size of the object downloaded by using `bcdc_query_geodata()` with `filter()` is `r paste0(format((1 - as.numeric(object.size(marmots2)) / as.numeric(object.size(all_sp_eco))) * 100, digits = 1), "%")` smaller than downloading the entire data set using `bcdc_get_data()` and filtering locally.
```
